### DISCUSSION TITLE
How to fix the out of bounds problem

### DISCUSSION THREAD

**Discussion by UlricQin** [ccfos/nightingale#1162] [General]
## 原因

Prometheus接收到了过于滞后的数据，就会报这个错误，通常是由于机器时间不准导致的

## 解法

升级夜莺到最新版本，在server.conf的第一行加个配置 `ForceUseServerTS = true`，这表示强制使用服务端时间作为监控数据的时间戳，不依赖客户端上报的监控数据的时间戳，这样可以解决99.9%的问题

**Comment by vivierer** [comment]
这个参数很管用，我用的是vm，客户端上使用了ntp授时，结果还是在n9e-server的日志上看到丢数据，报error out of bounds。开始以为时prometheus不能跑在vm上，换上了物理机，一样的问题。最后在最新版本的n9e中虽然升级日志说取消了ForceUseServerTS这个参数，我还是在server.conf中配置了。重启后报out of bounds的错误少了很多。然后我又调整了server.conf中write和http相关的时间参数，这个问题终于彻底消失了。

**Reply by xxqops** [reply]
您好。配置文件是否可以贴出来。我希望可以借鉴借鉴！

**Reply by lyitt1987** [reply]
能看下你的配置吗，谢谢
